refac(back): #1007 redefine cache module (#1010)
fluidattacks/makes: README.md

# Contents

> - [🦄 Makes](#-makes)

Reviewer: You need to run `/formatMarkdown` to get the proper contents table.

Types:

- cache:
  - extra: (attrsOf (cacheExtra))
  - readNixos (`bool`): Optional.
    Set to `true` in order to add https://cache.nixos.org as a read cache.
    Defaults to `true`.
- cacheExtra:

>   - enable (`str`): The current cache is enabled and the cache

Reviewer: ```suggestion
  - enable (`bool`): Read from cache.
```

    is read on the server.

>   - priority (`int`): the priority that the cache has when being read.

Reviewer: ```suggestion
  - priority (`int`): Priority when fetching derivations from this cache. 1 is highest.
```

>   - pubKey (`str`): public key to read the cache.

Reviewer: ```suggestion
  - pubKey (`str`): Public key of the cache server.
```

>   - token (`str`): the name of the environment variable that contains the

Reviewer: ```suggestion
  - token (`str`): Environment variable that contains the token for pushing to the cache.
```

>     token tu push the cache.

Reviewer: ```suggestion
    token to push the cache.
```

>     token tu push the cache.

Reviewer: ```suggestion

```

>   - type: (`enum [cachix | attic]`): the cache server type.

Reviewer: ```suggestion
  - type: (`enum [cachix | attic]`): Binary cache type. Can be Cachix or Attic (Agregar referencias a cada uno).
```

  - url (`str`):
    URL of the cache.

>   - write (`bool`): the cache is enabled to push the binary cache.

Reviewer: ```suggestion
  - write (`bool`): Enable pushing derivations to the cache. Requires `token`.
```

{
  cache = {
    readNixos = true;
    extra = {
      main = {
        enable = true;

>         priority = 20;

Reviewer: ```suggestion
        priority = 2;
```

        pubKey = "makes.cachix.org-1:zO7UjWLTRR8Vfzkgsu1PESjmb6ymy1e4OE9YfMmCQR4=";
        token = "CACHIX_AUTH_TOKEN";
        type = "nixos";
        url = "https://makes.cachix.org";
        write = true;
      };

>       local = {

Reviewer: Maybe moving this one first on the list so we sort based on priority?

        enable = true;

>         priority = 10;

Reviewer: ```suggestion
        priority = 1;
```
